### PR TITLE
[10.0] Migrate base import async

### PR DESCRIPTION
**Pull request by lmignon**
supersedes #33

### CODE REVIEW

**Comment by leorochael** [comment]
With this migration, `base_import_async` no longer depends on `connector`, so I don't know if it makes much sense to keep it in a repo called `connector-interfaces`.

How about we move this (and also `test_base_import_async`) into the `queue` repository itself? It's a small self-contained example that also serves to test if `queue_job` is properly configured.

@guewen, what do you think?

File: oca_dependencies.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 connector
```
Comment (leorochael) [inline oca_dependencies.txt]: I guess `connector` needs to be removed from `oca_dependencies.txt`, right?

**Comment by guewen** [comment]
> How about we move this (and also test_base_import_async) into the queue repository itself? It's a small self-contained example that also serves to test if queue_job is properly configured.

Agreed

File: base_import_async/models/base_import_import.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+from odoo.addons.queue_job.job import job, related_action
+from odoo.addons.queue_job.exception import FailedJobError
+
+# options defined in base_import/import.js
+OPT_HAS_HEADER = 'headers'
+OPT_SEPARATOR = 'separator'
+OPT_QUOTING = 'quoting'
+OPT_ENCODING = 'encoding'
+# options defined in base_import_async/import.js
+OPT_USE_CONNECTOR = 'use_connector'
```
Comment (guewen) [inline base_import_async/models/base_import_import.py]: would it be worth to rename to `OPT_USE_QUEUE`?

**Review by guewen — COMMENTED** [review]
Just a question, otherwise LGTM.
Thanks a lot!

File: base_import_async/models/base_import_import.py (hunk 2 above)
Comment (lmignon) [reply base_import_async/models/base_import_import.py]: @guewen good catch. I'll rename. Tests are also migrated now.

**Comment by lmignon** [comment]
@guewen @sbidoul can we merge this one?

**Review by sbidoul — APPROVED** [review]

**Comment by sbidoul** [comment]
@lmignon yes, we use it everywhere, works perfectly. We can move it to oca/queue in 11 or 12.

**Pull request merged**